server/models/Notification.ts

```diff
@@ -218,6 +226,65 @@ class Notification extends Model<
   public get pixelUrl() {
     return `${env.URL}/api/notifications.pixel?token=${this.pixelToken}&id=${this.id}`;
   }
+
+  /**
+   * Returns the past message ids which are used to setup the thread chain in email clients.
+   *
+   * @param notification Notification for which the past notifications are fetched - used for determining the properties that form a thread.
+   * @returns An array of email message ids that form a thread.
+   */
+  public static async emailReferences(
+    notification: Notification
+  ): Promise<string[] | undefined> {
+    if (!isEmailThreadSupportedNotification(notification.event)) {
+      return;
+    }
+
+    const events = getEmailThreadEventGroup(notification.event);
+
+    if (!events) {
+      return;
+    }
+
+    const prevNotifications: Notification[] = [];
+
+    await this.findAllInBatches<Notification>(
+      {
+        attributes: ["id"],
+        where: {
+          id: {
+            [Op.ne]: notification.id,
+          },
+          event: {
+            [Op.in]: events,
+          },
+          documentId: notification.documentId,
+          userId: notification.userId,
+        },
+        order: [["createdAt", "ASC"]],
+        offset: 0,
+        limit: 100,
```

Comment:
This changed to `batchLimit` on `main` here https://github.com/outline/outline/pull/7476

Reply:
Created #7511 for a bug related to this.
I'll push the review changes after #7511 is merged - Tests would fail otherwise.